src/main.rs

```diff
@@ -300,64 +300,73 @@ fn display_dispatch_error(err: DispatchError) -> String {
-    match result {
-        Err(err) => match err.downcast_ref::<UserErr>() {
-            Some(err) => match err {
-                UserErr::MentionedUserNotFound => {
-                    let _ = msg.reply_error(&ctx, "No user found with that name").await;
-                }
-                UserErr::InvalidUsage(usage) => {
-                    let _ = msg.reply_error(&ctx, format!("Usage: {}", usage)).await;
-                }
-                UserErr::Other(issue) => {
-                    let _ = msg.reply_error(&ctx, format!("Error: {}", issue)).await;
-                }
-            },
-            None => match err.downcast::<serenity::Error>() {
-                Ok(err) => {
-                    let err = *err;
-                    tracing::warn!(
-                        error.command_name = %command_name,
-                        error.message = %err,
-                        "Serenity error [handling {}]: {} ({:?})",
-                        command_name,
-                        &err,
-                        &err
-                    );
-                    match err {
-                        serenity::Error::Http(err) => {
-                            if let serenity::http::error::Error::UnsuccessfulRequest(res) = *err {
-                                if res.status_code == serenity::http::StatusCode::NOT_FOUND
-                                    && res.error.message.to_lowercase().contains("unknown user")
-                                {
-                                    let _ = msg.reply_error(&ctx, "User not found").await;
-                                } else {
-                                    let _ = msg.reply_error(&ctx, "Something went wrong").await;
-                                }
-                            }
-                        }
-                        serenity::Error::Model(err) => {
-                            let _ = msg.reply_error(&ctx, err).await;
-                        }
-                        _ => {
-                            let _ = msg.reply_error(&ctx, "Something went wrong").await;
-                        }
-                    }
-                }
-                Err(err) => {
-                    let _ = msg.reply_error(&ctx, "Something went wrong").await;
-                    tracing::warn!(
-                        error.command_name = %command_name,
-                        error.message = %err,
-                        "Internal error [handling {}]: {} ({:#?})",
-                        command_name,
-                        &err,
-                        &err
-                    );
+    let err = if let Err(err) = result { err } else { return };
+    // handle UserErr
+    if let Some(err) = err.downcast_ref::<UserErr>() {
+        match err {
+            UserErr::MentionedUserNotFound => {
+                msg.reply_error(&ctx, "No user found with that name")
+                    .await
+                    .ok();
```

Comment:
This is really not better than `let _ =` imo. No need to turn the result into an Option. Less expressive.